docs/rfcs/0007-flow-modules.md

```diff
@@ -0,0 +1,295 @@
+# Flow Modules
+
+* Date: 2023-01-27
+* Author: Matt Durham @mattdurham
+* PR: [grafana/agent#2898](https://github.com/grafana/agent/pull/2898)
+* Status: Draft
+
+[Formatted Link for ease of user](https://github.com/grafana/agent/blob/rfc_modules/docs/rfcs/0007-flow-modules.md)
+
+## RFC Goals
+
+* Explain the use cases of modules
+* Explain what modules are
+* Go over possible syntax for modules
+* Go over pros and cons of modules
+
+## Summary
+
+One of the primary goals for the production usage of Agent Flow is parity with the static subsystem. One of the features of the static subsystem is [scraping service](https://github.com/grafana/agent/blob/main/docs/sources/configuration/scraping-service.md). Scraping service allows a user to run a fleet of agents and have thousands of scrape configurations distributed among running Agents. Through discussions within the Agent Team, we did not want to limit dynamically loading content to only scrape configs but allow most components to be loaded and used.
+
+During this time the Agent team saw a lot of potential in modules outside of scraping service. Packaging up sets of components for specific workflows, publishing common use cases and allowing better usage for internal customers in the Agent as a Service model.
```

Comment:
I don't want to do too much feedback on formatting for my first review pass, but I'd like to see this be more clear that you're creating a new term and defining it on the next sentence:

```suggestion
During this time the Agent team saw a lot of potential in the idea of "modules." Modules package up sets of components for specific workflows, publishing common use cases and allowing better usage for internal customers in the Agent as a Service model.
```

Reply:
+1, this also makes the Goals section underneath much clearer as to what it's trying to communicate